### PR TITLE
[doc] Consistently use title and headings

### PR DESCRIPTION
**Pull request by imphil**
* Add a "title" attribute to make the page title show up in the
  generated TOC. (The first heading isn't used there unfortunately.)
* The title serves as "h1" heading, use "h2" headings (`##`) and below
  within the document.

### CODE REVIEW

File: doc/project/rfc_process.md [hunk 1]
```diff
@@ -1,4 +1,7 @@
-# OpenTitan RFC Process
+---
+aliases: [/doc/project/hw_stages/]
```
Comment (towoe) [inline doc/project/rfc_process.md:2]: Did not try to build it locally, but is it intended to have the same alias here as in "development_stages.md"?

**Review by rswarbrick — APPROVED** [review]
Ah, neat. I didn't know how `title` worked here. Thanks for doing this.

**Review by towoe — APPROVED** [review]
Looks much better with those changes.
Setting the alias again will result in "hw_stages" to go to rfc, so I guess this was copy paste error.
Other than that LGTM.

**Comment by imphil** [comment]
Yes, the alias was a copy-paste error, which is fixed in the latest revision.

**Pull request merged**